recipes/EOBS/recipe.py

```diff
@@ -0,0 +1,46 @@
+"""Author: Norland Raphael Hagen - 08-09-2022
+Pangeo-Forge recipe for E-OBS data (E-OBS: High-resolution gridded min and max air temperature, relative humidity, sea-level pressure and rainfall data for Europe & Northern Africa)"""  # noqa: E501
+
+
+from pangeo_forge_recipes.patterns import ConcatDim, FilePattern, MergeDim
+from pangeo_forge_recipes.recipes import XarrayZarrRecipe
+
+# Filename Pattern Inputs
+target_chunks = {'time': 40}
+dataset_version = 'v23.1e'
+grid_res = '0.1'
+subset_inputs = {'time': 700}
+
+
+def make_filename(time: str, variable: str) -> str:
+    return f'https://knmi-ecad-assets-prd.s3.amazonaws.com/ensembles/data/Grid_{grid_res}deg_reg_ensemble/{variable}_ens_mean_{grid_res}deg_reg_{dataset_version}.nc'  # noqa: E501
```

Comment:
@norlandrhagen, it appears apache-beam doesn't work well with globally defined variables such as `grid_res`.  Could you move these into the function body of `make_filename`?


```python
  File "apache_beam/runners/common.py", line 1417, in apache_beam.runners.common.DoFnRunner.process
  File "apache_beam/runners/common.py", line 624, in apache_beam.runners.common.SimpleInvoker.invoke_process
  File "/usr/local/lib/python3.9/dist-packages/apache_beam/transforms/core.py", line 1956, in <lambda>
  File "/usr/local/lib/python3.9/dist-packages/pangeo_forge_recipes/executors/beam.py", line 40, in exec_stage
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pangeo_forge_recipes/recipes/xarray_zarr.py", line 155, in cache_input
    fname = config.file_pattern[input_key]
  File "/srv/conda/envs/notebook/lib/python3.9/site-packages/pangeo_forge_recipes/patterns.py", line 219, in __getitem__
    fname = self.format_function(**format_function_kwargs)
  File "/tmp/tmpis_skwvg/recipes/EOBS/recipe.py", line 16, in make_filename
NameError: name 'grid_res' is not defined [while running 'Start|cache_input|Reshuffle_000|prepare_target|Reshuffle_001|store_chunk|Reshuffle_002|finalize_target|Reshuffle_003/cache_input/Execute-ptransform-56']
```

Reply:
Yeah totally. Thanks for looking into this @andersy005